packages/apps/quality/src/screens/ControlEntryDetailsScreen.js

```diff
@@ -0,0 +1,90 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2024 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, {useCallback} from 'react';
+import {
+  Button,
+  HeaderContainer,
+  Screen,
+  ScrollList,
+  useThemeColor,
+} from '@axelor/aos-mobile-ui';
+import {useDispatch, useSelector, useTranslator} from '@axelor/aos-mobile-core';
+import {ControlEntryHeader, ControlEntrySampleCard} from '../components';
+import {searchControlEntrySample} from '../features/controlEntrySampleSlice';
+
+const ControlEntryDetailsScreen = ({route}) => {
+  const {controlEntryId} = route.params;
+
+  const dispatch = useDispatch();
+  const I18n = useTranslator();
+  const Colors = useThemeColor();
+
+  const {
+    controlEntrySampleList,
+    loadingControlEntrySampleList,
+    moreLoading,
+    isListEnd,
+  } = useSelector(state => state.controlEntrySample);
+
+  const fetchControlEntryAPI = useCallback(
+    (page = 0) => {
+      dispatch(
+        searchControlEntrySample({
+          page: page,
+          controlEntryId: controlEntryId,
+        }),
+      );
+    },
+    [dispatch, controlEntryId],
+  );
+
+  return (
+    <Screen
+      removeSpaceOnTop
+      fixedItems={
+        <Button
+          title={I18n.t('Quality_MarkAsCompleted')}
+          iconName="check-lg"
+          color={Colors.successColor}
+        />
```

Comment:
This can be a new component, this way the onPress logic will be contain in it and not on the screen
Also, it should be in primaryColor